test/Microsoft.Extensions.Primitives.Tests/StringSegmentTest.cs

```diff
@@ -292,6 +295,61 @@ public void StringSegment_EqualsString_Invalid()
             Assert.False(result);
         }
 
+        public static TheoryData<StringSegment, StringComparison, bool> DefaultEqualsStringSegmentData
+        {
+            get
+            {
+                // candidate / comparer / expected result
+                return new TheoryData<StringSegment, StringComparison, bool>()
+                {
+                    { default(StringSegment), StringComparison.Ordinal, true },
```

Comment:
Other tests might do this, but it's much nicer to split out the `equals` and `not equals` scenarios in to two separate tests. In this case, move the last scenario `{ new StringSegment("Hello, World!", 1, 4), StringComparison.Ordinal, false }` to a separate test along the lines of `StringSegmentDoesNotEqualNonDefaultStringSegments`. This should change the test data to `TheoryData<StringSegment>`